x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_event_log_list_table.tsx

```diff
@@ -171,18 +182,14 @@ export const RuleEventLogListTable = <T extends RuleEventLogListOptions>(
   });
 
   const spacesData = useSpacesData();
-  const accessibleSpaceIds = useMemo(
-    () => (spacesData ? [...spacesData.spacesMap.values()].map((e) => e.id) : []),
-    [spacesData]
-  );
-  const areMultipleSpacesAccessible = useMemo(
-    () => accessibleSpaceIds.length > 1,
-    [accessibleSpaceIds]
-  );
-  const namespaces = useMemo(
-    () => (showFromAllSpaces && spacesData ? accessibleSpaceIds : undefined),
-    [showFromAllSpaces, spacesData, accessibleSpaceIds]
-  );
+  const { onShowAllSpacesChange, canAccessMultipleSpaces, namespaces } = useMultipleSpaces({
```

Comment:
let's move the `useSpacesData` call above into the hook since it's not being used anywhere else, same with  `actions_connectors` event log table. We won't have to pass in `spacesData` into the hook.

Reply:
Resolved in this commit b7c17038b4dd4966cecf06b30f2e43cd4b932275